src/main/java/com/paneedah/weaponlib/tile/CustomTileEntityRenderer.java

```diff
@@ -20,30 +31,52 @@ public CustomTileEntityRenderer(ModelBase model, ResourceLocation textureResourc
+     *
+     * @param tileEntity    The tile entity being rendered.
+     * @param posX         The x position for rendering.
+     * @param posY         The y position for rendering.
+     * @param posZ         The z position for rendering.
+     * @param partialTicks  Partial ticks for smoother rendering.
+     * @param destroyStage  The destroy stage of the block.
+     * @param alpha        The alpha value for transparency.
+     */
     @Override
-    public void render(CustomTileEntity<?> tileEntity, double posX, double posY, double posZ, float partialTicks, int destroyStage, float alpha) {
+    public void render(T tileEntity, double posX, double posY, double posZ,
+                       float partialTicks, int destroyStage, float alpha) {
         GL11.glPushMatrix();
-        this.bindTexture(textureResource);
+        bindTexture(textureResource);
+        setupRenderingTransformations(tileEntity, posX, posY, posZ);
+
+        // Call the positioning consumer to apply additional transformations if necessary
+        positioning.accept(tileEntity);
+
+        // Render the model
+        model.render(null, 0f, 0f, 0f, 0f, 0f, MODEL_RENDER_SCALE);
+
+        GL11.glPopMatrix();
+    }
 
-        //  GL11.glEnable(GL12.GL_RESCALE_NORMAL);
+    /**
+     * Sets up the rendering transformations based on tile entity's position and orientation.
+     *
+     * @param tileEntity The tile entity being rendered.
+     * @param posX      The x position for rendering.
+     * @param posY      The y position for rendering.
+     * @param posZ      The z position for rendering.
+     */
+    private void setupRenderingTransformations(T tileEntity, double posX, double posY, double posZ) {
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-        GL11.glTranslatef((float) posX, (float) posY + 1.0F, (float) posZ + 1.0F);
-        GL11.glScalef(1.0F, -1.0F, -1.0F);
+        GL11.glTranslatef((float) posX, (float) posY + TRANSLATE_Y, (float) posZ + TRANSLATE_Y);
+        GL11.glScalef(SCALE_FACTOR, -SCALE_FACTOR, -SCALE_FACTOR);
         GL11.glTranslatef(0.5F, 0.5F, 0.5F);
-        GL11.glRotatef(90f * tileEntity.getSide(), 0, 1f, 0);
-        GL11.glRotatef(-90f, 0, 1f, 0);
 
-        GL11.glTranslatef(-0.5F, -0.5F, -0.5F);
+        // Rotate based on the side of the tile entity
+        GL11.glRotatef(ROTATION_ANGLE * tileEntity.getSide(), 0, 1f, 0);
+        GL11.glRotatef(-ROTATION_ANGLE, 0, 1f, 0);
 
+        GL11.glTranslatef(-0.5F, -0.5F, -0.5F);
         GL11.glTranslatef(0F, -0.5F, 0F);
-
-        //System.out.println("Rendering side " + ((CustomTileEntity<?>) tileEntity).getSide());
-//        GL11.glRotatef(-90f * ((CustomTileEntity<?>) tileEntity).getSide(), 0, 1f, 0);
-        positioning.accept(tileEntity);
-        //GL11.glEnable(GL11.GL_CULL_FACE);
-        model.render(null, 0f, 0f, 0f, 0f, 0f, 0.0625f);
-
-        //  GL11.glDisable(GL12.GL_RESCALE_NORMAL);
-        GL11.glPopMatrix();
     }
```

Comment:
Please keeps comments for now, until I rewrite that part of rendering, it helps figure out what the hell was going on